packages/automated-dispute/src/providers/protocolProvider.ts

```diff
@@ -288,40 +293,197 @@ export class ProtocolProvider implements IProtocolProvider {
     }
 
     /**
-     * Gets a list of events between two blocks.
+     * Decodes the log data for a specific event.
      *
-     * @param {bigint} _fromBlock - The starting block number.
-     * @param {bigint} _toBlock - The ending block number.
-     * @returns {Promise<EboEvent<EboEventName>[]>} A list of EBO events.
+     * @param eventName - The name of the event to decode.
+     * @param log - The log object containing the event data.
+     * @returns The decoded log data as an object.
+     * @throws {Error} If the event name is unsupported or if there's an error during decoding.
      */
-    async getEvents(_fromBlock: bigint, _toBlock: bigint): Promise<EboEvent<EboEventName>[]> {
-        // TODO: implement actual method.
-        //
-        // We should decode events using the corresponding ABI and also "fabricate" new events
-        // if for some triggers there are no events (e.g. dispute window ended)
-        const eboRequestCreatorEvents: EboEvent<EboEventName>[] = [];
-
-        const oracleEvents = [
-            {
-                name: "ResponseDisputed",
-                blockNumber: 3n,
-                logIndex: 1,
-                requestId: "0x01" as RequestId,
+    private decodeLogData<TEventName extends EboEventName>(
+        eventName: TEventName,
+        log: Log,
+    ): DecodedLogArgsMap[TEventName] {
+        let abi;
+        switch (eventName) {
+            case "RequestCreated":
+                abi = eboRequestCreatorAbi;
+                break;
+            case "ResponseProposed":
+                abi = oracleAbi;
+                break;
+            case "ResponseDisputed":
+                abi = oracleAbi;
+                break;
+            case "DisputeStatusUpdated":
+                abi = oracleAbi;
+                break;
+            case "DisputeEscalated":
+                abi = oracleAbi;
+                break;
+            case "OracleRequestFinalized":
+                abi = oracleAbi;
+                break;
```

Comment:
```suggestion
            case "ResponseProposed":
            case "ResponseDisputed":
            case "DisputeStatusUpdated":
            case "DisputeEscalated":
            case "OracleRequestFinalized":
                abi = oracleAbi;
                break;
```
idk if you like fallthrough or not xd